packages/odata-query-objects/src/operation/QFunction.ts

```diff
@@ -19,7 +20,7 @@ function compileUrlParams(params: FunctionParams | undefined, notEncoded: boolea
   const queryParams: string[] = [];
   const pathParams = Object.entries(params).map(([key, value]) => {
     [key, value] = notEncoded ? [key, value] : [encodeURIComponent(key), encodeURIComponent(value)];
-    const isComplex = value.startsWith("{") || value.startsWith("[");
```

Comment:
with the aforementioned change you can stick to the previous expression with `value.startsWith(...)`